src/ConnectedMode/Hotspots/ServerHotspotStoreUpdater.cs

```diff
@@ -0,0 +1,111 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.ComponentModel.Composition;
+using System.Linq;
+using System.Threading.Tasks;
+using SonarLint.VisualStudio.ConnectedMode.Helpers;
+using SonarLint.VisualStudio.Core;
+using SonarLint.VisualStudio.Integration;
+using SonarQube.Client;
+
+namespace SonarLint.VisualStudio.ConnectedMode.Hotspots
+{
+    /// <summary>
+    /// Fetches Hotspots from the server and updates the server hotspot store
+    /// </summary>
+    public interface IServerHotspotStoreUpdater : IDisposable
+    {
+        /// <summary>
+        /// Fetches all available hotspots from the server and updates the <see cref="IServerHotspotStore"/>
+        /// </summary>
+        Task UpdateAllServerHotspotsAsync();
+    }
+
+    [Export(typeof(IServerHotspotStoreUpdater))]
+    [PartCreationPolicy(CreationPolicy.Shared)]
+    internal sealed class ServerHotspotStoreUpdater : IServerHotspotStoreUpdater
+    {
+        [ImportingConstructor]
+        public ServerHotspotStoreUpdater(ISonarQubeService sonarQube,
+            IServerHotspotStore serverHotspotStore,
+            IServerQueryInfoProvider serverQueryInfoProvider,
+            ICancellableActionRunner actionRunner,
+            IThreadHandling threadHandling,
+            ILogger logger)
+        {
+            this.sonarQube = sonarQube;
+            this.serverHotspotStore = serverHotspotStore;
+            this.serverQueryInfoProvider = serverQueryInfoProvider;
+            this.actionRunner = actionRunner;
+            this.threadHandling = threadHandling;
+            this.logger = logger;
+        }
+
+        private readonly ISonarQubeService sonarQube;
+        private readonly IServerHotspotStore serverHotspotStore;
+        private readonly IServerQueryInfoProvider serverQueryInfoProvider;
+        private readonly ICancellableActionRunner actionRunner;
+        private readonly IThreadHandling threadHandling;
+        private readonly ILogger logger;
+
+        public async Task UpdateAllServerHotspotsAsync()
+        {
+            await threadHandling.SwitchToBackgroundThread();
```

Comment:
TBH I just copied the behavior from the SuppressedIssueStoreUpdater. Should I just create a ticket to fix both of them?

Reply:
> TBH I just copied the behavior from the SuppressedIssueStoreUpdater.

That's the sort of thing it's important to call in a PR comment to help reviewers. Knowing that a PR is based on something that has already been reviewed might affect the amount of time and effort that needs to be put into reviewing the "new" code. Also, if a PR is based closely on existing code, it's important to call out in PR comments where the new code differs and why.

Here, I'm guessing both classes are relying on the fact they are being called in "fire and forget" way i.e. the caller triggers them but doesn't care about waiting for the result, so there is no code in the caller that will continue executing on a different thread. It's not a safe assumption as we have no idea how the class will be used in the future.

Ticket - yes, please do. It's not high-priority but it would be better to do the right thing.